multiqc/core/ai.py

```diff
@@ -0,0 +1,764 @@
+        )
+
+    def interpret_report_full(self, report_content: str) -> Optional[InterpretationResponse]:
+        def send_request() -> requests.Response:
+            return requests.post(
+                f"{config.seqera_api_url}/internal-ai/query",
+                headers={"Authorization": f"Bearer {self.api_key}"},
+                json={
+                    "system_message": PROMPT_FULL,
+                    "message": report_content,
+                    "tags": ["multiqc", f"multiqc_version:{config.version}"],
+                    "response_schema": {
+                        "name": "Interpretation",
+                        "description": "Interpretation of a MultiQC report",
+                        "input_schema": {
+                            "type": "object",
+                            "required": ["summary"],
+                            "properties": {
+                                key: {
+                                    "type": "string",
+                                    "description": value.description,
+                                    **({"default": value.default} if value.default is None else {}),
+                                }
+                                for key, value in InterpretationOutput.model_fields.items()
+                            },
+                        },
+                    },
+                },
+            )
+
+        if config.verbose:
+            response = send_request()
+        else:
+            response = run_with_spinner("ai", "Summarizing report with AI...", send_request)
+
+        if response.status_code != 200:
+            msg = f"Failed to get a response from Seqera: {response.status_code} {response.text}"
+            logger.error(msg)
+            if config.strict:
+                raise RuntimeError(msg)
+            return None
+
+        response_dict = response.json()
+        uuid = response_dict.get("uuid")
+        generation: Dict[str, str] = response_dict.get("generation")
+        return InterpretationResponse(
+            uuid=uuid,
+            interpretation=InterpretationOutput(**generation),
+            model=self.model or "claude-3-5-sonnet-latest",
+        )
+
+
+def get_llm_client() -> Optional[Client]:
+    if not config.ai_summary:
+        return None
+
+    if config.ai_provider == "seqera":
+        api_key = os.environ.get("SEQERA_ACCESS_TOKEN", os.environ.get("TOWER_ACCESS_TOKEN"))
+        if not api_key:
+            logger.error(
+                "config.ai_summary is set to true, and config.ai_provider is set to 'seqera', "
+                "but Seqera access token is not set. "
+                "Please set the SEQERA_ACCESS_TOKEN / TOWER_ACCESS_TOKEN environment variable "
+                "or change config.ai_provider"
+            )
+            return None
+        return SeqeraClient(config.ai_model, api_key)
+
+    if config.ai_provider == "anthropic":
+        api_key = os.environ.get("ANTHROPIC_API_KEY")
```

Comment:
Similar to above, would be nice having debug log messages saying that these env vars were found / used.